l2geth/core/state_processor.go

```diff
@@ -81,17 +86,12 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 // for the transaction, gas used and an error if the transaction failed,
 // indicating the block was invalid.
 func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, error) {
-	var msg Message
-	var err error
-	if !vm.UsingOVM {
-		msg, err = tx.AsMessage(types.MakeSigner(config, header.Number))
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		decompressor := config.StateDump.Accounts["OVM_SequencerEntrypoint"]
-		msg, err = AsOvmMessage(tx, types.MakeSigner(config, header.Number), decompressor.Address, header.GasLimit)
-		if err != nil {
+	msg, err := tx.AsMessage(types.MakeSigner(config, header.Number))
+	if err != nil {
```

Comment:
I think the diff from upstream here should be behind a `vm.UsingOVM` flag to make it obvious that its a diff from upstream